### ISSUE TITLE
Performance motivations for call tags?

### ISSUE THREAD

**Issue by tlively** [WebAssembly/call-tags#5]
In https://github.com/WebAssembly/gc/pull/243#issuecomment-934620861 and in other recent discussions, call tags have been raised as a potential solution for various performance problems that come up when compiling to Wasm GC. The call tags explainer describes call tags primarily as a mechanism for enforcing program invariants rather than performance, so it would be good to have the performance motivations written up as well.

**Comment by RossTate** [comment]
Good point. We recently implemented the low-level feature (though not in WebAssembly) and used it to implement an object-oriented language. Our experiments found it both performs extremely well and is very flexible, which is why it's been coming up. I'll put in a PR outlining some examples of how we've made use of them (and giving intuition as to why it performs well in those examples).
